src/OnboardingSPA/components/SiteGenError/index.js

```diff
@@ -0,0 +1,149 @@
+import { useViewportMatch } from '@wordpress/compose';
+import { useEffect } from '@wordpress/element';
+import { useSelect, useDispatch } from '@wordpress/data';
+import getContents from './contents';
+import { Button, Fill } from '@wordpress/components';
+import { store as nfdOnboardingStore } from '../../store';
+import CommonLayout from '../Layouts/Common';
+import OrbAnimation from '../OrbAnimation';
+import { SITEGEN_FLOW, DEFAULT_FLOW } from '../../data/flows/constants';
+import { validateFlow } from '../../data/flows/utils';
+import { resolveGetDataForFlow } from '../../data/flows';
+import { useNavigate } from 'react-router-dom';
+import {
+	FOOTER_SITEGEN,
+	FOOTER_END,
+	HEADER_SITEGEN,
+	pluginDashboardPage,
+} from '../../../constants';
+
+const SiteGenSiteError = () => {
+	const navigate = useNavigate();
+	const {
+		setIsHeaderEnabled,
+		setSidebarActiveView,
+		setHeaderActiveView,
+		setDrawerActiveView,
+		setHideFooterNav,
+		setIsHeaderNavigationEnabled,
+		updateAllSteps,
+		updateTopSteps,
+		updateRoutes,
+		updateDesignRoutes,
+		updateInitialize,
+		setCurrentOnboardingData,
+		updateSiteGenErrorStatus,
+	} = useDispatch( nfdOnboardingStore );
+
+	useEffect( () => {
+		setHideFooterNav( true );
+		setIsHeaderEnabled( true );
+		setSidebarActiveView( false );
+		setHeaderActiveView( HEADER_SITEGEN );
+		setIsHeaderNavigationEnabled( true );
+		setDrawerActiveView( false );
+	} );
+
+	const { brandConfig, currentData } = useSelect( ( select ) => {
+		return {
+			brandConfig: select( nfdOnboardingStore ).getNewfoldBrandConfig(),
+			currentData:
+				select( nfdOnboardingStore ).getCurrentOnboardingData(),
+		};
+	} );
+	const isLargeViewport = useViewportMatch( 'small' );
+
+	const content = getContents();
+	const oldFlow = window.nfdOnboarding?.oldFlow
+		? window.nfdOnboarding.oldFlow
+		: DEFAULT_FLOW;
+
+	const switchFlow = ( newFlow ) => {
+		if ( ! validateFlow( brandConfig, newFlow ) ) {
+			return false;
+		}
+		const currentFlow = window.nfdOnboarding.currentFlow;
+		const getData = resolveGetDataForFlow( newFlow );
+		const data = getData();
+		updateAllSteps( data.steps );
+		updateTopSteps( data?.topSteps );
+		updateRoutes( data.routes );
+		updateDesignRoutes( data?.designRoutes );
+		if ( SITEGEN_FLOW !== currentFlow ) {
+			window.nfdOnboarding.oldFlow = currentFlow;
+		}
+
+		window.nfdOnboarding.currentFlow = newFlow;
+		currentData.activeFlow = newFlow;
+		setCurrentOnboardingData( currentData );
+		if ( SITEGEN_FLOW !== newFlow ) {
+			updateInitialize( true );
+		}
+		navigate( data.steps[ 1 ].path );
+	};
+	const handleRetry = ( ) => {
+		updateSiteGenErrorStatus( false );
+		window.location.reload();
+	};
```

Comment:
We should prevent a reload of the tab. Setting the state to false should just re-render the original component along with its default behaviour.

Reply:
Changes to `currentData` should re-trigger the relevant API calls, let's add that in a `useEffect`.

Reply:
Done